sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -3521,6 +3523,25 @@ class SQLQuerySuite extends QueryTest with SharedSparkSession with AdaptiveSpark
             |""".stripMargin), Row(1))
     }
   }
+
+  test("SPARK-32002: Support Extract value from nested ArrayStruct") {
+    withTempView("rows") {
+      val df = spark.read
+        .json(Seq(
+          """{"a": [{"b": [{"c": [1,2]}]}]}""",
+          """{"a": [{"b": [{"c": [1]}, {"c": [2]}]}]}""",
+          """{"a":[{}]}""").toDS())
+      df.createOrReplaceTempView("nest")
+
+      checkAnswer(sql(
+        """
+          |SELECT a.b.c FROM nest
```

Comment:
Can we add deeper cases? Also, you can simplify the test cases, for example, as below:

```scala
val df = spark.range(10).select(array(struct(array(struct(id)).alias(col1))).alias(col0))
df.selectExpr("col0.col1.id")
```